### DISCUSSION TITLE
What is the proper typing for a Store instance?

### DISCUSSION THREAD

**Discussion by syropian** [vuejs/pinia#499] [Help and Questions] [answered]
Note: I'm currently using `2.0.0-alpha.17`

I'm working on a small custom composable thats purpose is to take a prop passed to a component, and set the store data with that prop's value:

```ts
import { watch } from 'vue'
import type { Store } from 'pinia'

export const useSyncValueToStore = <
  TStore extends Store,
  TKey extends keyof TStore
>(
  propGetter: () => TStore[TKey],
  store: TStore,
  key: TKey
): void => {
  watch(
    propGetter,
    propValue => {
      store[key] = propValue
    },
    {
      immediate: true,
    }
  )
}
```

That would get used like so:
```ts
useSyncValueToStore(() => props.user, userStore, 'user')
```

So basically: Take `props.user`, go into the `userStore` and set the value of the `user` state to the prop's value.

Unfortunately, TS is complaining about `userStore`:
```
Argument of type 'Store<"user", { user: Nullable<User>; }, GettersTree<{ user: Nullable<User>; }>, unknown>' is 
not assignable to parameter of type 'Store<string, StateTree, GettersTree<StateTree>, ActionsTree>'.
```

So what's the correct way to type the store instance in this case?

**Comment by posva** [comment, marked as answer]
I think this is a bug

**Reply by syropian** [reply]
Thanks @posva ! If that's the case I will just `@ts-ignore` it until it's fixed 😄

**Reply by posva** [reply]
use `// @ts-expect-error` so it fails once you can remove it and force you to remove it 💯

**Reply by posva** [reply]
Published as alpha-18
